lib/improver/tests/nowcasting/optical_flow/test_OpticalFlow.py

```diff
@@ -473,6 +474,40 @@ def test_values(self):
         self.assertAlmostEqual(np.mean(vmat), 0.121514428331)
 
 
+class Test_zero_advection_velocities_warning(IrisTest):
+    """Test the zero_advection_velocities_warning."""
```

Comment:
I wonder whether there should be another test here for when there is a zero in the array but it makes up less than 10% of numbers? That would just check that no warnings are raised in that instance?

Reply:
I've added a couple of unit tests. One, as you suggest, for where there are zeroes, but there are fewer zeroes than the threshold, and for where the threshold is modified.